tt_eager/tt_dnn/op_library/eltwise_binary/multi_core/eltwise_binary_op_multi_core.cpp

```diff
@@ -29,6 +29,10 @@ operation::ProgramWithCallbacks eltwise_binary_multi_core(const Tensor &a, const
     tt::DataFormat dst_cb_data_format = tt_metal::datatype_to_dataformat_converter(output.get_dtype());
     uint32_t dst_single_tile_size = tt_metal::detail::TileSize(dst_cb_data_format);
 
+
+    tt::DataFormat interim_cb_format = tt::DataFormat::Float16_b;
+    uint32_t interim_single_tile_size = tt_metal::detail::TileSize(interim_cb_format);
+
```

Comment:
If this is needed for only specific ops, then this should only be done for those ops or else you may cause unneeded degradation of perf.

I think you should add a mapping of op type -> interm data type, where default is input dtype if unspecified.